musicbrainz/get_genre.py

```diff
@@ -0,0 +1,31 @@
+"""
+Script for retrieving the "genre" dumps since they are not available online.
+"""
+import time
+import requests
+import pandas as pd
+
+URL = "https://musicbrainz.org/ws/2/genre/all"
+PARAMS = dict(fmt="json", limit="50", offset="0")
+
+data = []
+
+for i in range(0, 1947, 50):
+    PARAMS["offset"] = i
+    resp = requests.get(url=URL, params=PARAMS, timeout=100)
+    temp = resp.json()
+
+    try:
+        data += dict(temp)["genres"]
+    except KeyError:
```

Comment:
Sometimes, the request returns an error. This is to see the error message and not interrupting the requests.

Reply:
Right, but what error does it return. Is it an error that we should be mitigating in some way?

Reply:
It returns a dictionary, the same format as a normal response, but the only key is "error", not "genre".

Reply:
Why does it do that? When it does that do we just not get those genres? 

What I am getting at is that if there is a predictable error that is occurring, then it makes sense to have something in the code that mitigates this error or helps one know what to do in case of that error.

Reply:
> Why does it do that? When it does that do we just not get those genres?

I don't really get the pattern. Sometimes it happens and sometimes it does not. My way in this case is just avoid the interruption that it might cause.

Reply:
Is it some sort of timeout? 

What do you do with that set of genres? Do you just not include them?

Reply:
I think so. If error message appears, then I just run the script again to try again.

Reply:
This is the dictionary that error occurred.
`{'error': 'Your requests are exceeding the allowable rate limit. Please see http://wiki.musicbrainz.org/XMLWebService for more information.'}`

Reply:
Yeah, ok...so the error happens because you are requesting more frequently than the rate limit they have imposed...so you could:
- request less frequently
- when handling an exception, slow down the rate of your requests

Reply:
- check the status code of the HTTP requests -- that will let you know if the request succeeded or not (if it was `200` then yes; anything else, no.)
 - If you do receive a [`503`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503) you can catch that and re-try the request after waiting. 
 - [set a friendly `User-Agent` header](https://wiki.musicbrainz.org/MusicBrainz_API/Rate_Limiting#How_can_I_be_a_good_citizen_and_be_smart_about_using_the_Web_Service?)

Reply:
See here: https://requests.readthedocs.io/en/latest/user/quickstart/#custom-headers

Reply:
I know how your code works. I’m saying you’re going to be spamming them until you get rate limited, then you’re going to get hard limited.

why not follow their documentation and keep the number of all your requests to under 50/sec?